daemon/cluster/executor/container/controller.go

```diff
@@ -448,14 +475,19 @@ func (r *controller) Logs(ctx context.Context, publisher exec.LogPublisher, opti
 		return errors.Wrap(err, "failed getting container logs")
 	}
 
+	// use r.Task which is concurrency safe
+	task, tErr := r.Task()
+	if tErr != nil {
+		return errors.Wrap(err, "error getting controller task")
```

Comment:
oh jeez that's such a dumb oversight by me, i'm sorry. i've fixed it. i blame the post-dockercon haze.